### PR TITLE
Custom image name in the CSV generator

### PR DESCRIPTION
**Pull request by tiraboschi**
Let the user specify a custom image name
running the CSV genrator.

Fixes: https://bugzilla.redhat.com/1757798

### CODE REVIEW

File: pkg/components/components.go [hunk 1]
```diff
@@ -63,8 +63,8 @@ func (ai *AddonsImages) FillDefaults() *AddonsImages {
 	return ai
 }
 
-func GetDeployment(version string, operatorVersion string, namespace string, repository string, tag string, imagePullPolicy string, addonsImages *AddonsImages) *appsv1.Deployment {
-	image := fmt.Sprintf("%s/%s:%s", repository, Name, tag)
+func GetDeployment(version string, operatorVersion string, namespace string, repository string, name string, tag string, imagePullPolicy string, addonsImages *AddonsImages) *appsv1.Deployment {
```
Comment (phoracek) [inline pkg/components/components.go]: Name of what? deployment/container/image?

File: pkg/components/components.go [hunk 2]
```diff
@@ -63,8 +63,8 @@ func (ai *AddonsImages) FillDefaults() *AddonsImages {
...
+	image := fmt.Sprintf("%s/%s:%s", repository, name, tag)
```
Comment (phoracek) [inline pkg/components/components.go]: Here you use `name` for the image, in the rest of the PR you use it for container. What is the relation between the two?

**Comment by phoracek** [comment]
@tiraboschi is it really enough to make this change for operator image? Don't you need it for operands too?

**Comment by tiraboschi** [comment]
@phoracek 
without this we are currently getting:
```yaml
---
apiVersion: operators.coreos.com/v1alpha1
kind: ClusterServiceVersion
metadata:
  name: cluster-network-addons-operator.2.1.0
  namespace: placeholder
  annotations:
    capabilities: "Full Lifecycle"
    categories: "Networking"
    alm-examples: |
      [
        {
          "apiVersion":"networkaddonsoperator.network.kubevirt.io/v1alpha1",
          "kind":"NetworkAddonsConfig",
          "metadata": {
            "name":"cluster"
          },
          "spec": {
            "multus":{},
            "linuxBridge":{},
            "kubeMacPool": {
              "rangeStart": "02:00:00:00:00:00",
              "rangeEnd": "FD:FF:FF:FF:FF:FF"
            },
            "nmstate":{},
            "ovs": {},
            "imagePullPolicy": "IfNotPresent"
          }
        }
      ]
    description: Additional networking components for Kubernetes
spec:
  displayName: Cluster Network Addons
  description: Deploy additional networking components for Kubernetes
  keywords:
    - Networking
    - Multus
    - CNI
    - macpool
    - SR-IOV
    - Bridge
    - nmstate
    - KubeVirt
    - Virtualization
  version: 2.1.0
  minKubeVersion: 1.10.0
  maturity: alpha

  replaces: cluster-network-addons-operator.2.0.0

  maintainers:
    - name: KubeVirt project
      email: kubevirt-dev@googlegroups.com
  provider:
    name: KubeVirt project
  links:
    - name: Cluster Network Addons Operator
      url: https://github.com/kubevirt/cluster-network-addons-operator
  icon: []
  labels:
    alm-owner-kubevirt: cluster-network-addons
    operated-by: cluster-network-addons
  selector:
    matchLabels:
      alm-owner-kubevirt: cluster-network-addons
      operated-by: cluster-network-addons
  installModes:
    - type: OwnNamespace
      supported: true
    - type: SingleNamespace
      supported: true
    - type: MultiNamespace
      supported: true
    - type: AllNamespaces
      supported: true
  install:
    strategy: deployment
    spec:
      permissions:
        - serviceAccountName: cluster-network-addons-operator
          rules:
            - apiGroups:
              - ""
              resources:
              - pods
              - configmaps
              verbs:
              - get
              - list
              - watch
              - create
              - patch
              - update
              - delete
            - apiGroups:
              - apps
              resources:
              - deployments
              - replicasets
              verbs:
              - get
              - list
              - watch
              - create
              - patch
              - update
              - delete

      clusterPermissions:
        - serviceAccountName: cluster-network-addons-operator
          rules:
            - apiGroups:
              - security.openshift.io
              resourceNames:
              - privileged
              resources:
              - securitycontextconstraints
              verbs:
              - get
              - list
              - watch
            - apiGroups:
              - operator.openshift.io
              resources:
              - networks
              verbs:
              - get
              - list
              - watch
            - apiGroups:
              - networkaddonsoperator.network.kubevirt.io
              resources:
              - networkaddonsconfigs
              verbs:
              - get
              - list
              - watch
            - apiGroups:
              - '*'
              resources:
              - '*'
              verbs:
              - '*'

      deployments:
        - name: cluster-network-addons-operator
          spec:
            replicas: 1
            selector:
              matchLabels:
                name: cluster-network-addons-operator
            strategy:
              type: Recreate
            template:
              metadata:
                labels:
                  name: cluster-network-addons-operator
              spec:
                containers:
                - env:
                  - name: MULTUS_IMAGE
                    value: registry-proxy.engineering.redhat.com/rh-osbs/openshift-ose-multus-cni:v4.2
                  - name: LINUX_BRIDGE_IMAGE
                    value: registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-cnv-containernetworking-plugins:v2.1.0-6
                  - name: LINUX_BRIDGE_MARKER_IMAGE
                    value: registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-bridge-marker:v2.1.0-3
                  - name: NMSTATE_HANDLER_IMAGE
                    value: registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-kubernetes-nmstate-handler-rhel8:v2.1.0-7
                  - name: OVS_CNI_IMAGE
                    value: registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-ovs-cni-plugin:v2.1.0-9
                  - name: OVS_MARKER_IMAGE
                    value: registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-ovs-cni-marker:v2.1.0-9
                  - name: KUBEMACPOOL_IMAGE
                    value: registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-kubemacpool:v2.1.0-8
                  - name: OPERATOR_IMAGE
                    value: registry-proxy.engineering.redhat.com/rh-osbs/cluster-network-addons-operator:v2.1.0-12
                  - name: OPERATOR_NAME
                    value: cluster-network-addons-operator
                  - name: OPERATOR_VERSION
                    value: v2.1.0-12
                  - name: OPERATOR_NAMESPACE
                    valueFrom:
                      fieldRef:
                        fieldPath: metadata.namespace
                  - name: OPERAND_NAMESPACE
                    valueFrom:
                      fieldRef:
                        fieldPath: metadata.namespace
                  - name: POD_NAME
                    valueFrom:
                      fieldRef:
                        fieldPath: metadata.name
                  - name: WATCH_NAMESPACE
                  image: registry-proxy.engineering.redhat.com/rh-osbs/cluster-network-addons-operator:v2.1.0-12
                  imagePullPolicy: IfNotPresent
                  name: cluster-network-addons-operator
                  resources: {}
                serviceAccountName: cluster-network-addons-operator

  customresourcedefinitions:
    owned:
      - name: networkaddonsconfigs.networkaddonsoperator.network.kubevirt.io
        version: v1alpha1
        group: networkaddonsoperator.network.kubevirt.io
        kind: NetworkAddonsConfig
        displayName: Cluster Network Addons
        description: Cluster Network Addons
```
where `registry-proxy.engineering.redhat.com/rh-osbs/cluster-network-addons-operator:v2.1.0-12` is for sure wrong because it should be `registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization/cluster-network-addons-operator:v2.1.0-12`.
All the other images pull URL are fine.

Still not really sure about other names.

**Comment by tiraboschi** [comment]
with this we obtain:
```bash
[stirabos@t470s cluster-network-addons-operator]$ docker run --rm --entrypoint=/usr/bin/csv-generator quay.io/kubevirt/cluster-network-addons-operator:latest  --container-name=container-native-virtualization-cluster-network-addons-operator  --version=2.1.0 --version-replaces=2.0.0 --operator-version=v2.1.0-12 --namespace=openshift-cnv --container-prefix=registry-proxy.engineering.redhat.com/rh-osbs --container-tag=v2.1.0-12 --multus-image=registry-proxy.engineering.redhat.com/rh-osbs/openshift-ose-multus-cni:v4.2 --linux-bridge-cni-image=registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-cnv-containernetworking-plugins:v2.1.0-6 --linux-bridge-marker-image=registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-bridge-marker:v2.1.0-3 --kubemacpool-image=registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-kubemacpool:v2.1.0-8 --nm-state-handler-image=registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-kubernetes-nmstate-handler-rhel8:v2.1.0-7 --ovs-cni-image=registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-ovs-cni-plugin:v2.1.0-9 --ovs-marker-image=registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-ovs-cni-marker:v2.1.0-9 --image-pull-policy=IfNotPresent --dump-crds
apiVersion: operators.coreos.com/v1alpha1
kind: ClusterServiceVersion
metadata:
  name: cluster-network-addons-operator.2.1.0
  namespace: placeholder
  annotations:
    capabilities: "Full Lifecycle"
    categories: "Networking"
    alm-examples: |
      [
        {
          "apiVersion":"networkaddonsoperator.network.kubevirt.io/v1alpha1",
          "kind":"NetworkAddonsConfig",
          "metadata": {
            "name":"cluster"
          },
          "spec": {
            "multus":{},
            "linuxBridge":{},
            "kubeMacPool": {
              "rangeStart": "02:00:00:00:00:00",
              "rangeEnd": "FD:FF:FF:FF:FF:FF"
            },
            "nmstate":{},
            "ovs": {},
            "imagePullPolicy": "IfNotPresent"
          }
        }
      ]
    description: Additional networking components for Kubernetes
spec:
  displayName: Cluster Network Addons
  description: Deploy additional networking components for Kubernetes
  keywords:
    - Networking
    - Multus
    - CNI
    - macpool
    - SR-IOV
    - Bridge
    - nmstate
    - KubeVirt
    - Virtualization
  version: 2.1.0
  minKubeVersion: 1.10.0
  maturity: alpha

  replaces: cluster-network-addons-operator.2.0.0

  maintainers:
    - name: KubeVirt project
      email: kubevirt-dev@googlegroups.com
  provider:
    name: KubeVirt project
  links:
    - name: Cluster Network Addons Operator
      url: https://github.com/kubevirt/cluster-network-addons-operator
  icon: []
  labels:
    alm-owner-kubevirt: cluster-network-addons
    operated-by: cluster-network-addons
  selector:
    matchLabels:
      alm-owner-kubevirt: cluster-network-addons
      operated-by: cluster-network-addons
  installModes:
    - type: OwnNamespace
      supported: true
    - type: SingleNamespace
      supported: true
    - type: MultiNamespace
      supported: true
    - type: AllNamespaces
      supported: true
  install:
    strategy: deployment
    spec:
      permissions:
        - serviceAccountName: cluster-network-addons-operator
          rules:
            - apiGroups:
              - ""
              resources:
              - pods
              - configmaps
              verbs:
              - get
              - list
              - watch
              - create
              - patch
              - update
              - delete
            - apiGroups:
              - apps
              resources:
              - deployments
              - replicasets
              verbs:
              - get
              - list
              - watch
              - create
              - patch
              - update
              - delete

      clusterPermissions:
        - serviceAccountName: cluster-network-addons-operator
          rules:
            - apiGroups:
              - security.openshift.io
              resourceNames:
              - privileged
              resources:
              - securitycontextconstraints
              verbs:
              - get
              - list
              - watch
            - apiGroups:
              - operator.openshift.io
              resources:
              - networks
              verbs:
              - get
              - list
              - watch
            - apiGroups:
              - networkaddonsoperator.network.kubevirt.io
              resources:
              - networkaddonsconfigs
              verbs:
              - get
              - list
              - watch
            - apiGroups:
              - '*'
              resources:
              - '*'
              verbs:
              - '*'

      deployments:
        - name: cluster-network-addons-operator
          spec:
            replicas: 1
            selector:
              matchLabels:
                name: cluster-network-addons-operator
            strategy:
              type: Recreate
            template:
              metadata:
                labels:
                  name: cluster-network-addons-operator
              spec:
                containers:
                - env:
                  - name: MULTUS_IMAGE
                    value: registry-proxy.engineering.redhat.com/rh-osbs/openshift-ose-multus-cni:v4.2
                  - name: LINUX_BRIDGE_IMAGE
                    value: registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-cnv-containernetworking-plugins:v2.1.0-6
                  - name: LINUX_BRIDGE_MARKER_IMAGE
                    value: registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-bridge-marker:v2.1.0-3
                  - name: NMSTATE_HANDLER_IMAGE
                    value: registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-kubernetes-nmstate-handler-rhel8:v2.1.0-7
                  - name: OVS_CNI_IMAGE
                    value: registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-ovs-cni-plugin:v2.1.0-9
                  - name: OVS_MARKER_IMAGE
                    value: registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-ovs-cni-marker:v2.1.0-9
                  - name: KUBEMACPOOL_IMAGE
                    value: registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-kubemacpool:v2.1.0-8
                  - name: OPERATOR_IMAGE
                    value: registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-cluster-network-addons-operator:v2.1.0-12
                  - name: OPERATOR_NAME
                    value: cluster-network-addons-operator
                  - name: OPERATOR_VERSION
                    value: v2.1.0-12
                  - name: OPERATOR_NAMESPACE
                    valueFrom:
                      fieldRef:
                        fieldPath: metadata.namespace
                  - name: OPERAND_NAMESPACE
                    valueFrom:
                      fieldRef:
                        fieldPath: metadata.namespace
                  - name: POD_NAME
                    valueFrom:
                      fieldRef:
                        fieldPath: metadata.name
                  - name: WATCH_NAMESPACE
                  image: registry-proxy.engineering.redhat.com/rh-osbs/container-native-virtualization-cluster-network-addons-operator:v2.1.0-12
                  imagePullPolicy: IfNotPresent
                  name: cluster-network-addons-operator
                  resources: {}
                serviceAccountName: cluster-network-addons-operator

  customresourcedefinitions:
    owned:
      - name: networkaddonsconfigs.networkaddonsoperator.network.kubevirt.io
        version: v1alpha1
        group: networkaddonsoperator.network.kubevirt.io
        kind: NetworkAddonsConfig
        displayName: Cluster Network Addons
        description: Cluster Network Addons
---
apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
  name: networkaddonsconfigs.networkaddonsoperator.network.kubevirt.io
spec:
  group: networkaddonsoperator.network.kubevirt.io
  names:
    kind: NetworkAddonsConfig
    listKind: NetworkAddonsConfigList
    plural: networkaddonsconfigs
    singular: networkaddonsconfig
  scope: Cluster
  subresources:
    status: {}
  validation:
    openAPIV3Schema:
      properties:
        apiVersion:
          type: string
        kind:
          type: string
        metadata:
          type: object
        spec:
          type: object
        status:
          type: object
  version: v1alpha1
  versions:
  - name: v1alpha1
    served: true
    storage: true
```

**Comment by phoracek** [comment]
@tiraboschi please address my comments on the code

**Comment by tiraboschi** [comment]
Verified:

```

File: pkg/components/components.go (hunk 1 above)
Comment (tiraboschi) [reply pkg/components/components.go]: image only

**Pull request merged**